### PR TITLE
[Variant Picker] Simplify swatch settings

### PR DESCRIPTION
**Pull request by alisterdev**
### PR Summary: 

Simplifying swatch settings by merging `options_with_swatches` setting into `swatch_shape` setting

### Why are these changes introduced?

Addressing UX feedback

### What approach did you take?

- Updated settings structure
- Simplified translations structure since we no longer require additional nesting

| Before | After |
| :----: | :---: |
|   image |  image |

### Visual impact on existing themes

More streamline settings for variant-picker

### Testing steps/scenarios

🎩 pill, dropdown, swatch, dropdown with swatch configs

It's probably helpful to review the code with whitespace changes hidden: https://github.com/Shopify/dawn/pull/3189/files?w=1

### Demo links

- [Store](url)
- [Editor](url)

### Checklist
- [x] Added PR summary for [release notes](https://themes.shopify.com/themes/dawn/styles/default#ReleaseNotes)
- [ ] Requested review from UX (Only for changes that are affecting the experience or perceivable visual details)
- [ ] Created a ticket for the [help.shopify.com](https://help.shopify.com) documentation team about updates to theme settings. (Internal-only task)
- [x] Followed [theme code principles](https://github.com/Shopify/dawn/blob/main/.github/CONTRIBUTING.md#theme-code-principles)
- [ ] Linted with [Theme Check](https://github.com/Shopify/theme-check)
- [ ] Tested on [mobile](https://shopify.dev/themes/store/requirements#mobile-browser-requirements)
- [ ] Tested on [multiple browsers](https://shopify.dev/themes/store/requirements#desktop-browser-requirements)
- [ ] Tested for [accessibility](https://shopify.dev/themes/best-practices/accessibility)

### CODE REVIEW

File: sections/featured-product.liquid [hunk 1]
```diff
@@ -807,67 +807,39 @@
...
-            {
-              "value": "button",
-              "label": "t:sections.featured-product.blocks.variant_picker.settings.options_with_swatches.picker_type.options__4.label"
-            }
-          ],
-          "default": "swatch"
+          "label": "t:sections.featured-product.blocks.variant_picker.settings.picker_type.label"
         },
         {
           "id": "swatch_shape",
```
Comment (brian-lyn) [inline sections/featured-product.liquid:827]: Should we change the id of this setting to `swatch` to align with the new name/purpose of the setting?

File: sections/main-product.liquid [hunk 2]
```diff
@@ -804,67 +804,39 @@
...
               "value": "square",
-              "label": "t:sections.main-product.blocks.variant_picker.settings.options_with_swatches.swatch_shape.options__2.label"
+              "label": "t:sections.main-product.blocks.variant_picker.settings.swatch_shape.options__2.label"
+            },
+            {
+              "value": "none",
+              "label": "t:sections.main-product.blocks.variant_picker.settings.swatch_shape.options__3.label"
             }
           ],
           "default": "circle"
```
[the remark this answers is not in the document]
Comment (j-wanita) [reply sections/main-product.liquid:842]: The benefit of having `"default": "circle"` is that it requires no configuration in Dawn for merchants to display swatches (assuming they've set up their options in admin correctly), making it a more seamless experience.

File: locales/en.default.schema.json [hunk 3]
```diff
@@ -850,44 +850,26 @@
...
-              "picker_type": {
-                "label": "Style",
-                "options__1": {
-                  "label": "Dropdown"
-                },
-                "options__2": {
-                  "label": "Pills"
-                }
+            "picker_type": {
+              "label": "Style",
```
Comment (ludoboludo) [inline locales/en.default.schema.json:854]: It used to be `Type` before and not `style`. Maybe we can switch that back to stay as it was 🤔

**Review by ludoboludo — APPROVED** [review]
Just one small change and other wise good. 

We still need to wait for translations to be back though, ill ask in the team to see if they can be prioritized :)

File: sections/featured-product.liquid (hunk 1 above)
Comment (brian-lyn) [reply sections/featured-product.liquid:827]: Lets keep it as is for now but if we are able to manually change this once we get translations I think we should

**Review by brian-lyn — APPROVED** [review]

**Comment by bakura10** [comment]
I'm trying to think on how architecturing our theme, and I can see that you are offering an option "None" for the swatch shape setting. If I understand it correctly, this allows a merchant to explicitly select swatch in the admin, but force them to not display as swatch, as that correct? What is the exact use case here? Wouldn't a merchant explicitly selecting swatch in the admin always want to show them as swatch?

**Comment by alisterdev** [comment]
>If I understand it correctly, this allows a merchant to explicitly select swatch in the admin, but force them to not display as swatch, as that correct? 

👋 Hey! With the latest changes announced at [Editions 2024](https://www.shopify.com/ca/editions/winter2024) merchants will be able to have colors or images associated with their product options. It is powered by metaobjects and is exposed in the API using the [product_option_value_swatch](https://shopify.dev/docs/api/liquid/objects/product_option_value_swatch) drop.

When a product option is connected to a color metaobject using this new flow, `product_option_value_swatch` will be populated with the relevant information. In dawn, we default to displaying a swatch when `product_option_value_swatch` has a value, but give merchants an option to disable swatch style using the "None" option.

So, there is no way to "enable/disable" swatches in the admin, just a way to associate color/image data with product options.

You can check how the whole flow will work here: https://www.shopify.com/ca/editions/winter2024/video/product-merchandising-updates

Hope this helps 🙌

**Pull request merged**